solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -300,12 +301,11 @@ contract InterchainAccountRouter is Router {
         bytes32 _sender,
         bytes calldata _message
     ) external payable override onlyMailbox {
-        (
-            bytes32 _owner,
-            bytes32 _ism,
-            CallLib.Call[] memory _calls,
-            bytes32 _salt
-        ) = InterchainAccountMessage.decode(_message);
+        InterchainAccountMessage.MessageType _messageType = InterchainAccountMessage
+                .messageType(_message);
+        bytes32 _owner = InterchainAccountMessage.owner(_message);
+        bytes32 _ism = InterchainAccountMessage.ism(_message);
+        bytes32 _salt = InterchainAccountMessage.salt(_message);
```

Comment:
should we adopt the `using InterchainAccountMessage for bytes` syntax?

Reply:
Done.